### PR TITLE
fix(plugin-meetings): remove StatsAnalyzer in sdk

### PR DESCRIPTION
**Pull request by antsukanova**
# COMPLETES # [SPARK-541329](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-541329)

## This pull request addresses

This PR addresses removing `StatsAnalazer` from SDK and importing it from `internal-media-core.` The WCME team already did one bug bash session to verify these changes, and now I would like to ask the SDK team to verify the code, too. This is a big change, but the code itself has not changed that much.

## by making the following changes

Next things were changed to make new StatsAnalazer work properly:

1. Rename `Event` To  `MediaConnectionEventNames`
2. Import  `StatsAnalyzer` with types from `internal-media-core`
3. Delete existing StatsAnalyzer/ utils and tests
4. Update `setupStatsAnalyzerEventHandlers` in meeting/index.ts
5. Fix types issues
6. Fix tests 

##  how to test

We have merged the `internal-media-core` [PR](https://sqbu-github.cisco.com/WebExSquared/webrtc-media-core/pull/252) with the StatsAnalzer, and it's available in the 2.9.0 version which includes here.

You'll need to pull these changes, do `yarn,` `yarn @all build:src,` and run the sample app as usual.

**_Please NOTE_**: you need to have `webrtc-media-core` `move_stats_analyzer` branch with the latest changes linked locally to SDK.

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [x] Internal code refactor

## The following scenarios were tested

WCME team conducted a bug bash session and tested different scenarios for sending metrics.
All of them were successful.
_**The following scenarios were tested:**_
- [x] streams array in the MQE look correct
- [x] NetworkQualityMonitor (from the SDK) still works with StatsAnalyzer.
- [x] All MQE data that was previously populated are still being populated.
- [x] All recent changes to StatsAnalyzer still work properly.
- [x] All tests have been updated and passed.
- [x] SDK plugin-meetings build and tests work properly.
- [x] SDK media-helpers build, and tests work properly.
- [x] SDK calling build and tests work properly.

### I certified that

- [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [x] I discussed changes with code owners prior to submitting this pull request
- [x] I have not skipped any automated checks
- [x] All existing and new tests passed
- [x] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: packages/@webex/media-helpers/package.json
Comment (antsukanova) [inline packages/@webex/media-helpers/package.json:25]: temporary for better testing

File: packages/@webex/plugin-meetings/src/meeting/index.ts
Comment (antsukanova) [inline packages/@webex/plugin-meetings/src/meeting/index.ts:6082]: Not sure if I correctly understand our intentions here. We can keep it in `options.networkType = options.data.networkType` or have it only in `payload` to avoid duplication. With that logic, I have been leaning towards the second option.

**Comment by brycetham** [comment]
Putting a note here as a reminder to also updated the package.json for the calling SDK.

File: packages/@webex/plugin-meetings/src/meeting/index.ts
Comment (antsukanova) [reply packages/@webex/plugin-meetings/src/meeting/index.ts:6082]: @brycetham Yeah, I thought the same, so I will keep it for now. I think I'll add a comment later for the SDK team to discuss this. Thanks

Comment (antsukanova) [reply packages/@webex/plugin-meetings/src/meeting/index.ts:6082]: @brycetham changed this part with networkType.

File: packages/@webex/plugin-meetings/test/unit/spec/stats-analyzer/index.js
Comment (antsukanova) [inline packages/@webex/plugin-meetings/test/unit/spec/stats-analyzer/index.js:3]: I have kept this file for the work test run command for now. Otherwise, deleing this now causes window `ReferenceError: window is not defined.`
It will be fixed separately

See details here webexteams://im?space=f2606c60-8ab2-11ec-be25-b7a3d486240d&message=bc2b0750-5ba6-11ef-994e-ad1d96cf0b70

File: packages/@webex/plugin-meetings/test/unit/spec/meeting/index.js
Comment (brycetham) [inline packages/@webex/plugin-meetings/test/unit/spec/meeting/index.js:7]: Ah, that's weird that we had both `internalMediaModule` and `InternalMediaCoreModule` previously. I would stick with the `InternalMediaCoreModule` name though, since that is the name of the imported module.

**Review by brycetham — COMMENTED** [review]
Looks good, left a small comment. I can approve once https://sqbu-github.cisco.com/WebExSquared/webrtc-media-core/pull/252 is merged and we resolve https://github.com/webex/webex-js-sdk/pull/3769#discussion_r1723680127.

File: packages/@webex/plugin-meetings/test/unit/spec/meeting/index.js
Comment (antsukanova) [reply packages/@webex/plugin-meetings/test/unit/spec/meeting/index.js]: @brycetham aligned naming for tests in `plugin-meetings`, `calling` and `media-helpers`

Thanks for that

File: packages/@webex/plugin-meetings/test/unit/spec/rtcMetrics/index.ts
Comment (antsukanova) [inline packages/@webex/plugin-meetings/test/unit/spec/rtcMetrics/index.ts:2]: This resolves an issue with tests "ReferenceError: window is not defined"

File: packages/@webex/plugin-meetings/src/meeting/index.ts
Comment (Kesari3008) [inline packages/@webex/plugin-meetings/src/meeting/index.ts:6424]: Why are we removing this? Is this not needed for meetings also ?

File: packages/calling/package.json
Comment (antsukanova) [reply packages/calling/package.json:40]: @Kesari3008 no, so far in description I put the purpose behind this change currently here, and I hope it makes sense to you

> this PR I created with local linking to make sure if someone wants to test in that's the fastest way. These changes will be removed after confirming that we are ready to merge this PR.

File: packages/calling/src/CallingClient/calling/call.test.ts
Comment (antsukanova) [reply packages/calling/src/CallingClient/calling/call.test.ts:4]: @Kesari3008 I wouldn't say that it's a must, but in one file, there were two imports with different names for `'@webex/internal-media-core'` in one file. I've changed how we mock `'@webex/internal-media-core'` for all test cases. We can discuss if you want to use another variable for this somewhere, but it feels right to align the naming if we want to make our lives easier with the codebase. This can be done separately if you feel it could be an issue. But so far, it's just a naming without logic change part, so it should be okay.

File: packages/@webex/plugin-meetings/src/meeting/index.ts
Comment (antsukanova) [reply packages/@webex/plugin-meetings/src/meeting/index.ts:6424]: @Kesari3008 this logic has been updated, so now we don't have to have `receiveSlotCallback` here.

You can see how it works now here:

WCME PR here: https://sqbu-github.cisco.com/CPaaS/web-client-media-engine/pull/392
internal-media-core PR here: https://sqbu-github.cisco.com/WebExSquared/webrtc-media-core/pull/234

File: packages/@webex/plugin-meetings/src/media/MediaConnectionAwaiter.ts
Comment (antsukanova) [reply packages/@webex/plugin-meetings/src/media/MediaConnectionAwaiter.ts:2]: Initially, we imported **Event** from '@webex/internal-media-core' as a media connection event object. However, we encountered a naming conflict with the StatAnalyzer in '@webex/internal-media-core', which also uses the same naming for its events. As a result, the codebase became inconsistent. While it is possible to keep **Event** without change, it feels odd to have different entities events exposed in the API of '@webex/internal-media-core'. I believe that making this change will ensure consistency in the API.

File: packages/@webex/media-helpers/test/unit/spec/webrtc-core.js
Comment (antsukanova) [reply packages/@webex/media-helpers/test/unit/spec/webrtc-core.js:16]: addressed here https://github.com/webex/webex-js-sdk/pull/3769#discussion_r1730858737

**Review by Kesari3008 — APPROVED** [review]
Ensure to revert the local linking before merge and amplify url and build should pass
Please upload logs for the testing done is possible to the PR for reference

**Review by sreenara — APPROVED** [review]

**Review by brycetham — APPROVED** [review]

**Review by edvujic — APPROVED** [review]

**Pull request merged**
